clang/test/AST/Interp/literals.cpp

```diff
@@ -31,6 +31,13 @@ static_assert(b, "");
 constexpr int one = true;
 static_assert(one == 1, "");
 
+constexpr bool b2 = bool();
```

Comment:
There are a lot of scalars: https://eel.is/c++draft/basic.types.general#9

Can we have a test for each case?